test/e2e/framework/timeouts.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import "time"
+
+// TimeoutContext contains timeout settings for several actions.
+type TimeoutContext struct {
+	// PodList is how long to wait for the pod to be listable.
+	PodList time.Duration
+
+	// PodStart is how long to wait for the pod to be started.
+	PodStart time.Duration
+
+	// PodStartShort is same as `PodStart` to wait for the pod to be started, but shorter.
+	// Use it case by case when we are sure pod start will not be delayed.
+	// minutes by slow docker pulls or something else.
+	PodStartShort time.Duration
+
+	// PodDelete is how long to wait for a pod to be deleted.
+	PodDelete time.Duration
+
+	// PodGet is how long to wait for a pod to be got.
+	PodGet time.Duration
+
+	// PodEvent is how much we wait for a pod event to occur.
+	PodEvent time.Duration
+
+	// ServiceStart is how long to wait for a service endpoint to be resolvable.
+	ServiceStart time.Duration
+
+	// Poll is how often to Poll pods, nodes and claims.
+	Poll time.Duration
```

Comment:
Does this need to be configurable? IMO it's about the frequency and IMO we can have one that makes everyone happy - once per 2s should be OK.

Reply:
Removed